admin/bootstrap.py

```diff
@@ -111,24 +105,22 @@ def is_missing_dependency() -> bool:
     This applies to existing Tails systems where `securedrop-setup` may not have been
     run recently.
     """
-    # apt-cache -q0 policy $dependency1 $dependency2 $dependency3 | grep "Installed: (none)"
+
     apt_query = f"apt-cache -q0 policy {APT_DEPENDENCIES_STR}".split(" ")
-    grep_command = ["grep", "Installed: (none)"]
 
     try:
         sdlog.info("Checking apt dependencies are installed")
-        apt_process = subprocess.Popen(apt_query, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        apt_query_result = subprocess.run(apt_query, capture_output=True, text=True)
 
-        grep_process = subprocess.Popen(
-            grep_command, stdin=apt_process.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-        )
+        # If any packages are marked as not installed, return 0
+        if "Installed: (none)" in apt_query_result.stdout:
+            return True
 
-        # Wait for the process to complete before checking the returncode
-        grep_process.communicate()
-        returncode = grep_process.returncode
+        # If any packages couldn't be found, it may point to an apt cache issue
+        if "Unable to locate package" in apt_query_result.stderr:
+            return True
 
-        # If the above command returns 0, then one or more packages are not installed.
-        return returncode == 0
+        return False
 
     except subprocess.CalledProcessError as e:
```

Comment:
This is no longer needed AFAICT, per https://docs.python.org/3.9/library/subprocess.html#subprocess.run it is only raised if check=True, which you're not passing (maybe you should be checking the return code? not sure).

Reply:
Thanks @legoktm! Good catch! Rather than removing the exception handling, I think we _want_ to be alerted to an instance where running apt returns a non-zero exit code, so the exception is handled appropriately (as it has been in the past).

I've tacked on a commit where I pass check=True, which I think should be all that's necessary to process that return code.

If that looks good to you, we can squash the commit. And if not, let me know how you think we should proceed and I'll adjust accordingly.